include/seqan3/core/char_operations/predicate_detail.hpp

```diff
@@ -344,19 +344,50 @@ struct is_in_alphabet_type : public char_predicate_base<is_in_alphabet_type<alph
 // is_char_type
 // ----------------------------------------------------------------------------
 
+/*!\brief Helper struct that constructs a seqan3::small_string.
+ * \ingroup stream
+ * \tparam char_v non-type template parameter with the value to construct a seqan3::small_string from.
+ *
+ * If char_v == EOF, the returned small_string contains "EOF", otherwise the small_string is constructed from char_v.
+ */
+template <int char_v>
+struct string_helper
+{
+       //!\brief Returns a small_string constructed from char_v if char_v != EOF and small_string{"EOF"} otherwise.
+       static constexpr small_string<1> get_string()
+       //!\cond
+               requires (char_v != EOF)
+       //!\endcond
+       {
+               return small_string{char_v};
+       }
+
+       //!\overload
+       static constexpr small_string<3> get_string()
+       //!\cond
+               requires (char_v == EOF)
+       //!\endcond
+       {
+               return small_string{"EOF"};
+       }
+};
+
 /*!\brief Parse condition that checks if a given value is equal to `char_v`.
  * \ingroup stream
  * \implements seqan3::detail::CharPredicate
- * \tparam alphabet_t non-type template parameter with the value that should be checked against.
+ * \tparam char_v non-type template parameter with the value that should be checked against.
  */
 template <int char_v>
 struct is_char_type : public char_predicate_base<is_char_type<char_v>>
 {
     static_assert(char_v == EOF || static_cast<uint64_t>(char_v) < 256, "TODO");
 
+    //!\brief Helper struct to handle char_v == EOF.
+    static string_helper<char_v> helper;
+
     //!\brief The message representing this condition.
     static constexpr auto msg = small_string{"is_char<'"} +
-                                small_string{char_v}      +
+                                helper.get_string()       +
```

Comment:
and `(char_v == EOF) ? small_string{"EOF"} : small_string<3>(char_v)` did not work?

Reply:
Nope, still `error: narrowing conversion of ‘-1’ from ‘int’ to ‘char’ inside { } [-Wnarrowing]`, because `small_string<3>(char_v)` needs to be valid in the ternary.

Reply:
```cpp
    static constexpr auto lam = [] () {
        if constexpr (char_v == EOF)
            return small_string{"EOF"};
        else
            return small_string{char_v};
    };
```

would work ( and then `lam()` instead of `helper.get_string()`).